internal/mode/static/nginx/config/maps_test.go

```diff
@@ -189,3 +188,94 @@ func TestBuildAddHeaderMaps(t *testing.T) {
 
 	g.Expect(maps).To(ConsistOf(expectedMap))
 }
+
+func TestExecuteStreamMaps(t *testing.T) {
+	g := NewWithT(t)
+	conf := dataplane.Configuration{
+		TLSPassthroughServers: []dataplane.Layer4VirtualServer{
+			{
+				Hostname:     "example.com",
+				Port:         8081,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "example.com",
+				Port:         8080,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "cafe.example.com",
+				Port:         8080,
+				UpstreamName: "backend2",
+			},
+		},
```

Comment:
let's include an SSL server here, but you can also make this test a bit simpler since you are testing the shared.Map structure in the next test.